src/XMakeBuildEngine/Evaluation/ProjectParser.cs

```diff
@@ -368,6 +389,53 @@ private ProjectItemElement ParseProjectItemElement(XmlElementWithLocation elemen
             return item;
         }
 
+        private void CheckMetadataName(string name, out bool isKnownAttribute, out bool isValidMetadataNameInAttribute)
+        {
+            if (!XmlUtilities.IsValidElementName(name))
+            {
+                isKnownAttribute = false;
+                isValidMetadataNameInAttribute = false;
+                return;
+            }
+
+            for (int i = 0; i < s_knownAttributesOnItem.Length; i++)
+            {
+                //  Case insensitive comparison so that mis-capitalizing an attribute like Include or Exclude results in an easy to understand
+                //  error instead of unexpected behavior
+                if (name == s_knownAttributesOnItem[i])
+                {
+                    isKnownAttribute = true;
+                    isValidMetadataNameInAttribute = false;
+                    return;
+                }
+                else if (name.Equals(s_knownAttributesOnItem[i], StringComparison.OrdinalIgnoreCase))
+                {
+                    isKnownAttribute = false;
+                    isValidMetadataNameInAttribute = false;
+                    return;
+                }
+            }
+
+            //  Resorve attributes starting with underscores in case we need to add more built-in attributes later
```

Comment:
Pretty much, unless we're OK with potentially breaking some people.  Using an underscore was the best thing I could think of... if you have a better idea I'd love to hear it.